akd/src/storage/memory.rs

```diff
@@ -558,6 +584,18 @@ impl Storage for AsyncInMemoryDbWithCache {
     }
 
     async fn batch_set(&self, records: Vec<DbRecord>) -> Result<(), StorageError> {
+        if records.is_empty() {
+            // nothing to do, save the cycles
+            return Ok(());
+        }
+
+        if self.is_transaction_active().await {
+            for record in records {
+                self.trans.set(&record).await;
+            }
+            return Ok(());
+        }
```

Comment:
It looks like there is a decent amount of duplication across the impl blocks for the traits. Maybe we can come back to this eventually with some work focused on clean up. Not a blocker for this diff by any means!